synapse/rest/client/room_batch.py

```diff
@@ -383,11 +383,20 @@ async def on_POST(
         # Prepend the insertion event to the start of the batch (oldest-in-time)
         events_to_create = [insertion_event] + events_to_create
 
+        # Also connect the historical event chain to floating state chain,
+        # which causes the HS to ask for the state at the start of
+        # the batch later.
+        prev_event_ids = [prev_event_id_for_state_chain]
```

Comment:
Connect our historical batch chain to the float state chain so we still get the floating benefit but also can resolve in the state and share `state_groups` across the batch.